### PR TITLE
[Publisher] Admin Panel: Clean Up

### LINKED ISSUE #1075: [Publisher] Admin Panel: Clean up
- [x] Reasonable responsiveness (#1105)
- [x] Styling adjustments (#1105)
- [x] Figure out how to better sync the loading of the fetched users/agencies (#1103)
- [x] Ensure proper error handling and messaging (#1105)
- [x] Update "System" to "Sector" (#1105)
- [x] Remove superagency system addition/removal -> checkmark update logic b/c it is no longer necessary (user doesn't select the superagency system anymore, the checkmark adds or removes it accordingly) (#1105)
- [x] Fix clash between filter checkmarks and search box in `AgencyProvisioningOverview` (#1105)

### PR DESCRIPTION
**Pull request by mxosman**
## Description of the change

Addresses the following clean up items:
- Styling clean up adjustments
- Update "System" to "Sector"
- Ensure proper error handling and messaging (when trying to create existing users/agencies)
- Remove superagency system addition/removal -> checkmark update logic b/c it is no longer necessary (user doesn't select the superagency system anymore, the checkmark adds or removes it accordingly)
- Fix clash between search box and filter checkmarks in `AgencyProvisioningOverview`

Here's a quick demo of the error handling for existing users/agencies:

https://github.com/Recidiviz/justice-counts/assets/59492998/37544267-f6d4-40c3-b423-223eb24dfab3

## Related issues

Closes #1075 

### Development

**This box MUST be checked by the submitter prior to merging**:
- [x] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [ ] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: publisher/src/components/AdminPanel/UserProvisioning.tsx [hunk 1]
```diff
@@ -256,24 +261,28 @@ export const UserProvisioning: React.FC<ProvisioningProps> = observer(
...
+              <Styled.NameDisplay>
+                {userProvisioningUpdates?.name || selectedUser?.name}
+              </Styled.NameDisplay>
+              <Styled.Subheader>
+                {userProvisioningUpdates?.email || selectedUser?.email}
+              </Styled.Subheader>
+              {selectedUser && (
+                <Styled.Subheader>ID {selectedUser?.id}</Styled.Subheader>
+              )}
+            </Styled.ModalHeader>
```
Comment (mxosman) [inline publisher/src/components/AdminPanel/UserProvisioning.tsx:285]: Just added a `<Styled.ModalHeader>` here - the rest is indentation.

File: publisher/src/components/AdminPanel/AgencyProvisioningOverview.tsx [hunk 2]
```diff
@@ -48,15 +48,29 @@ export const AgencyProvisioningOverview = observer(() => {
...
+    [agencies]
+  );
+  const superagencies = useMemo(
+    () => agencies.filter((agency) => agency.is_superagency),
+    [agencies]
+  );
+  const agenciesWithLiveDashboards = useMemo(
+    () => agencies.filter((agency) => agency.is_dashboard_enabled),
+    [agencies]
+  );
```
Comment (mxosman) [inline publisher/src/components/AdminPanel/AgencyProvisioningOverview.tsx:73]: Memoizing these to pass them into the useEffect dependency array (and avoid an infinite loop).

File: publisher/src/components/AdminPanel/AgencyProvisioning.tsx [hunk 3]
```diff
@@ -442,26 +444,30 @@ export const AgencyProvisioning: React.FC<ProvisioningProps> = observer(
...
+              </Styled.NameDisplay>
+              {selectedAgency && (
+                <Styled.Subheader>ID {selectedAgency?.id}</Styled.Subheader>
+              )}
 
-            {/* Toggle between Agency Information and Team Members & Roles */}
-            <TabbedBar options={settingOptions} />
+              {/* Toggle between Agency Information and Team Members & Roles */}
+              <TabbedBar options={settingOptions} />
+            </Styled.ModalHeader>
```
Comment (mxosman) [inline publisher/src/components/AdminPanel/AgencyProvisioning.tsx:470]: Just added a <Styled.ModalHeader> here - the rest is indentation.

File: publisher/src/components/AdminPanel/AdminPanel.styles.tsx [hunk 4]
```diff
@@ -69,7 +69,7 @@ export const SettingsBar = styled.div`
 `;
 
 export const ScrollableContainer = styled.div`
-  height: 61vh;
```
Comment (nichelle-hall) [inline publisher/src/components/AdminPanel/AdminPanel.styles.tsx:72]: what's vh?

**Review by nichelle-hall — APPROVED** [review]
LGTM ✨

File: publisher/src/components/AdminPanel/AdminPanel.styles.tsx (hunk 4 above)
Comment (mxosman) [reply publisher/src/components/AdminPanel/AdminPanel.styles.tsx:72]: TY so much, Nichelle! `vh` is the viewport height - so CSS can automatically calculate the height of the browser viewport. 61vh means 61% of the calculated viewport height - which was a hacky way to set the height of a container.

image

**Review by lilidworkin — APPROVED** [review]

**Pull request merged**